tenseal/tensors/ckksvector.cpp

```diff
@@ -37,8 +39,19 @@ CKKSVector::CKKSVector(const CKKSVector& vec) {
     this->ciphertext = vec.ciphertext;
 }
 
-size_t CKKSVector::size() { return this->_size; }
-size_t CKKSVector::ciphertext_size() { return this->ciphertext.size(); }
+CKKSVector::CKKSVector(const string& ctx, const string& vec) {
+    this->load_context(ctx);
+    this->load(vec);
+}
```

Comment:
The receiving machine will need to have multiple copies of the context in memory. It would be great if ctx is an actual TenSEALContext pointer object, that we only set without any deserialization, so that the receiving machine will deserialize the context, then link it to all the other vectors without replicating it.

I think a function `link_to_context` for vector might make sense here, to decouple the deserialization with this lonking logic, so maybe we deserialize normally, but the vector won't be operational until we link it to a context, does this make sense?

Reply:
I think an overload would be a better:
`CKKSVector::CKKSVector(shared_ptr<TenSEALContext> ctx, const string& vec)`
This will allow creating a vector using an exiting TenSEALContext as well as from a serialized TenSEALContext.

Reply:
We shouldn't create contexts per encrypted vector, context should be shared, so I think we shouldn't have a method for deserializing a context directly into a vector.